domain/ide/workspace/folder.go

```diff
@@ -142,20 +142,54 @@ func (f *Folder) DocumentDiagnosticsFromCache(file string) []snyk.Issue {
 }
 
 func (f *Folder) processResults(issues []snyk.Issue) {
-	var issuesByFile = map[string][]snyk.Issue{}
+	if len(issues) == 0 {
+		f.documentDiagnosticCache.ClearAll()
+		return
+	}
```

Comment:
This is a bug. 
This function can be called from a scan of a sub-folder/specific file, but the cache holds issues for all files/folders. and the results get cleared

Reply:
👍 this will be removed once #192 is merged, which will make this line redundant